src/classes/BusinessLogicServices/ShopOrderService.php

```diff
@@ -0,0 +1,335 @@
+     */
+    public function updateTrackingInfo($orderId, Shipment $shipment, array $trackingHistory)
+    {
+        if (!empty($shipment->trackingCodes)) {
+            $repository = new OrderRepository();
+            $repository->setTrackingNumber((int)$orderId, $shipment->trackingCodes[0]);
+        }
+    }
+
+    /**
+     * Sets order Packlink shipping status to an order with a given ID.
+     *
+     * @param string $orderId Shop order ID.
+     * @param string $shippingStatus Packlink shipping status.
+     *
+     * @throws \Packlink\BusinessLogic\Order\Exceptions\OrderNotFound When order for provided reference is not found.
+     */
+    public function updateShipmentStatus($orderId, $shippingStatus)
+    {
+        $repository = new OrderRepository();
+
+        /** @var ConfigurationService $configService */
+        $configService = ServiceRegister::getService(Configuration::CLASS_NAME);
+        $statusMappings = $configService->getOrderStatusMappings();
+
+        if (!array_key_exists($shippingStatus, $statusMappings)) {
+            Logger::logWarning(
+                TranslationUtility::__('Order status mapping not found.'),
+                'Integration'
+            );
+
+            return;
+        }
+
+        $repository->updateOrderState((int)$orderId, (int)$statusMappings[$shippingStatus]);
+    }
+
+    /**
+     * Fetches and returns system order by its unique identifier.
+     *
+     * @param string $orderId $orderId Unique order id.
+     *
+     * @return Order Order object.
+     *
+     * @throws \Logeecom\Infrastructure\ORM\Exceptions\QueryFilterInvalidParamException
+     * @throws \Logeecom\Infrastructure\ORM\Exceptions\RepositoryNotRegisteredException
+     * @throws \PrestaShopDatabaseException
+     * @throws \PrestaShopException
+     */
+    public function getOrderAndShippingData($orderId)
+    {
+        $order = new Order();
+        try {
+            $sourceOrder = $this->getOrder($orderId);
+        } catch (OrderNotFound $e) {
+            Logger::logWarning(TranslationUtility::__('Source order not found'), 'Integration');
+
+            return $order;
+        }
+
+        $currencyId = (int)$sourceOrder->id_currency;
+        $currency = \Currency::getCurrency($currencyId);
+
+        $order->setId($orderId);
+        $order->setCustomerId((int)$sourceOrder->id_customer);
+        $order->setCurrency($currency['iso_code']);
+        $order->setTotalPrice((float)$sourceOrder->total_paid_tax_incl);
+        $order->setBasePrice((float)$sourceOrder->total_paid_tax_excl);
+
+        $dropOffId = $this->getDropOffId($sourceOrder);
+
+        if ($dropOffId) {
+            $order->setShippingDropOffId($dropOffId);
+        }
+
+        $order->setShippingAddress($this->getAddress($sourceOrder));
+
+        $this->setOrderShippingDetails($order, $sourceOrder->id_carrier);
+        $order->setItems($this->getOrderItems($sourceOrder));
+
+        return $order;
```

Comment:
This logic could be implement in try stamen. I think is more clear

Reply:
Then, exception handling would be different, and that is not the intentend behavior.

Reply:
The exception handling is the same , if  $sourceOrder = $this->getOrder($orderId); fail then the exception was captured and not execute the code after this line .

Reply:
It is not the same if the code that is currently outside the try-catch block throws an exception. Anyhow, updated.